moto/__init__.py

```diff
@@ -171,22 +172,24 @@ def f(*args, **kwargs):
 mock_sdb = lazy_load(".sdb", "mock_sdb", boto3_name="sdb")
 
 
-def mock_all():
-    dec_names = [
-        d
-        for d in dir(sys.modules["moto"])
-        if d.startswith("mock_")
-        and not d.endswith("_deprecated")
-        and not d == "mock_all"
-    ]
+class mock_all(ContextDecorator):
+    def __init__(self):
+        self.mocks = []
+        for mock in dir(sys.modules["moto"]):
```

Comment:
Nitpick, but now that we’re not doing a try/except, we could switch back to list comprehension instead of a for loop. Similar to how it was written in the original commit (minus the X-ray client exclusion):

https://github.com/spulec/moto/blob/a7ae379a7d5c6934ad0ca8ea5257dce130ad59d4/moto/__init__.py#L177-L184

I’m fine with either - up to you what you find more readable.

Reply:
The for-loop is more readable IMO - the indentation helps understand better what's going on